auth/credentials/impersonate/idtoken.go

```diff
@@ -85,49 +90,45 @@ func NewIDTokenCredentials(opts *IDTokenOptions) (*auth.Credentials, error) {
 	}
 	var client *http.Client
 	var creds *auth.Credentials
-	if opts.Client == nil && opts.Credentials == nil {
+	if opts.Client == nil {
 		var err error
-		// TODO: test not signed jwt more
-		creds, err = credentials.DetectDefault(&credentials.DetectOptions{
-			Scopes:           []string{defaultScope},
-			UseSelfSignedJWT: true,
-		})
-		if err != nil {
-			return nil, err
+		if opts.Credentials == nil {
+			creds, err = credentials.DetectDefault(&credentials.DetectOptions{
+				Scopes:           []string{defaultScope},
+				UseSelfSignedJWT: true,
+			})
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			creds = opts.Credentials
```

Comment:
nit: you can set this value before the if and then check if it is nil after.

Reply:
Great suggestion, fixed in two files, thanks!